test/upgrade/run-upgrade.sh

```diff
@@ -5,129 +5,140 @@ set -vex
+    ")
+  if [ "${AWSACCOUNTTABLE}" == "aws_account" ]; then # It's the old style
+    psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
+        INSERT INTO aws_account (access_key_id, secret_access_key, region)
+        VALUES ('${AWS_ACCESS_KEY_ID}', '${AWS_SECRET_ACCESS_KEY}', 'us-east-1');
+      "
+  else
+    psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
+        INSERT INTO aws_credentials (access_key_id, secret_access_key)
+        VALUES ('${AWS_ACCESS_KEY_ID}', '${AWS_SECRET_ACCESS_KEY}');
+      "
+    psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
+        SELECT * FROM iasql_sync();
+      "
+    psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
+        SELECT * FROM default_aws_region('us-east-1');
+      "
   fi
-  sleep 1;
-  UPGRADECHECKCOUNT=$((${UPGRADECHECKCOUNT}-1))
-done
 
-# The check that the upgrade successfully loads up the new version for the account
-if [ "${ISUPGRADED}" == "false" ]; then
-  echo "Did not successfully upgrade!";
-  exit 2;
-fi
+  # Add one module that we can use to verify that everything actually comes back up
+  psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
+      SELECT * FROM iasql_install('aws_ec2');
+    "
 
-# Another loop to wait for the `aws_ec2` module to load
-EC2CHECKCOUNT=30
-EC2UPGRADED=false
-while [ ${EC2CHECKCOUNT} -gt 0 ]; do
-  AWSEC2MODULE=`psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -AXqtc "
-    SELECT name FROM iasql_module WHERE name like 'aws_ec2%'
-  "`
-  if [ "${AWSEC2MODULE}" == "aws_ec2@${LATESTVERSION}" ]; then
-    EC2UPGRADED=true
-    EC2CHECKCOUNT=0
-  fi
-  sleep 1;
-  EC2CHECKCOUNT=$((${EC2CHECKCOUNT}-1))
-done
+  # Shut down the database and engine, switch back to the current commit, and fire up a new engine
+  docker container stop $(basename ${PWD})_change_engine_1
+  docker container stop $(basename ${PWD})_postgresql_1
+  git checkout ${CURRENTGITSHA}
+  yarn docker-compose &
+  yarn wait-on http://localhost:8088/health/
 
-# The check that the upgrade successfully loads up the new version for the `aws_ec2` module
-if [ "${EC2UPGRADED}" == "false" ]; then
-  echo "Did not successfully upgrade!";
-  exit 3;
-fi
+  # Actually trigger the upgrade and loop until upgraded (or fail)
+  psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -c "
+      SELECT * FROM iasql_upgrade();
+    "
+  UPGRADECHECKCOUNT=30
+  ISUPGRADED=false
+  while [ ${UPGRADECHECKCOUNT} -gt 0 ]; do
+    AWSACCOUNTMODULE=$(psql postgres://postgres:test@127.0.0.1:5432/to_upgrade -AXqtc "
+        SELECT name FROM iasql_module WHERE name like 'aws_account%'
+      ")
+    if [ "${AWSACCOUNTMODULE}" == "aws_account@${LATESTVERSION}" ]; then
+      ISUPGRADED=true
+      UPGRADECHECKCOUNT=0
+    fi
+    sleep 1
+    UPGRADECHECKCOUNT=$((${UPGRADECHECKCOUNT} - 1))
+  done
+
+  # The check that the upgrade successfully loads up the new version for the account
+  if [ "${ISUPGRADED}" == "false" ]; then
```

Comment:
can we get a list of the modules that failed to upgrade? would be helpful

Reply:
well if one fails it will exit, so we will know exactly which one failed